src/server/worker/api_server.go

```diff
@@ -535,10 +536,19 @@ func (a *APIServer) downloadData(pachClient *client.APIClient, logger *taggedLog
 		if err := createSpoutFifo(outPath); err != nil {
 			return "", fmt.Errorf("mkfifo :%v", err)
 		}
-		_, err := pachClient.InspectFile(a.pipelineInfo.Pipeline.Name, a.pipelineInfo.OutputBranch, "marker")
-		if err != nil && strings.Contains(err.Error(), "not found") {
-			if err := puller.Pull(pachClient, filepath.Join(dir, "marker"), a.pipelineInfo.Pipeline.Name, a.pipelineInfo.OutputBranch, "/marker", false, false, concurrency, nil, ""); err != nil {
-				return "", err
+		if a.pipelineInfo.Spout.Marker != "" {
+			// check if we have a marker file
+			_, err := pachClient.InspectFile(a.pipelineInfo.Pipeline.Name, ppsconsts.SpoutMarkerBranch, a.pipelineInfo.Spout.Marker)
+			// if not, then we need to pull it
+			if err != nil && strings.Contains(err.Error(), "not found") {
```

Comment:
Better to implement this as an `IsErrNotFound` function or import one from somewhere.